### DISCUSSION TITLE
Where are all the start commands and url params？

### DISCUSSION THREAD

**Discussion by huangJW7** [gitpod-io/openvscode-server#430] [Q&A]
Hey, I want to use some commands and url params to simplify some operations. 

e.g.  `http://localhost:3000/?folder=/home/workspace`  can open the folder instead of chick `open folder`.

e.g. `--user-extension-dir` can Specify the extra extensions.

I want to open a terminal by default when open the webpage. Can this be achieved with url params? Where can I find all the params and usage?

By the way, `/.openvscode-server/bin/remote-cli/code` have many functions such as `diff`,`merge`,`add`. But it says `Command is only available in WSL or inside a Visual Studio Code terminal.` How can I use it in shell?

**Comment by jeanp413** [comment]
Hi @huangJW7, you can find cli options [here](https://github.com/gitpod-io/openvscode-server/blob/a3c6c7380b181ea33714fbbbab41b65ba4df6ebe/src/vs/server/node/server.cli.ts#L56-L76) and server start script options [here](https://github.com/gitpod-io/openvscode-server/blob/a3c6c7380b181ea33714fbbbab41b65ba4df6ebe/src/vs/server/node/serverEnvironmentService.ts#L16-L86)

> I want to open a terminal by default when open the webpage

I don't think it's possible with the cli nor the server script options, internally vscode will remember the previous state if you open it on the same browser, one way would be to create an extension that creates and focus a new terminal

>  But it says Command is only available in WSL or inside a Visual Studio Code terminal. How can I use it in shell?

cli is meant to be used inside the vscode terminals, not in external shells

**Reply by sullemanhossam** [reply]
What about url paramaters :)

**Reply by LinuxSuRen** [reply]
You can see it from https://github.com/gitpod-io/openvscode-server/blob/a3c6c7380b181ea33714fbbbab41b65ba4df6ebe/src/vs/server/node/serverEnvironmentService.ts#L44
